### PR TITLE
initial linux support

### PR DESCRIPTION
**Pull request by marler8997**
With this commit I'm able to build SDL for linux and use it to build/run https://github.com/allyourcodebase/VVVVVV on my Ubuntu machine.

Note that I added a couple changes to VVVVVV for linux on this branch: https://github.com/allyourcodebase/VVVVVV/commit/48f01b8c38a5498c19382ce5afb7e723e5784cf1, namely just to disable a couple of the render drivers that VVVVVV doesn't support.  Since the linux build is customizing the config, I added some build options that allow the application to enable/disable a few parts of it like the software render driver.

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -11,12 +11,13 @@ pub fn build(b: *std.Build) void {
         .optimize = optimize,
     });
 
-    lib.addIncludePath(b.path("include"));
     lib.addCSourceFiles(.{ .files = &generic_src_files });
     lib.defineCMacro("SDL_USE_BUILTIN_OPENGL_DEFINITIONS", "1");
     lib.linkLibC();
+    var use_pregenerated_config = false;
```
Comment (andrewrk) [inline build.zig]: `const use_pregenerated_config = switch (t.os.tag) { ...`

as a rule of thumb, `var` should be used pretty much only in loops.

File: build.zig [hunk 2]
```diff
@@ -60,16 +63,92 @@ pub fn build(b: *std.Build) void {
...
+        defer files.deinit();
+
+        // copy all headers except the pregenerated SDL_config.h and SDL_revision.h
+        // so we can use our configured headers instead (SDL's CMakeLists.txt does
+        // the same thing)
+        const write_files = b.addWriteFiles();
+        inline for (static_headers) |header| {
+            _ = write_files.addCopyFile(b.path("include/" ++ header), header);
+        }
+        lib.addIncludePath(write_files.getDirectory());
```
Comment (andrewrk) [inline build.zig]: it seems sus to do this rather than having 2 include paths, but I guess if that's what SDL does with their build then it's fine.

**Review by andrewrk — CHANGES_REQUESTED** [review]
Nice!

This is fine but I guess since I made a concrete suggestion in the review, I'll press the "request changes" button.

Other than that, it looks good to merge :+1:

File: build.zig (hunk 2 above)
Comment (marler8997) [reply build.zig]: that sounds like a way better idea, gonna try this

**Comment by marler8997** [comment]
The idea to split the headers into two different directories worked like a charm.  I've updated the commit to do this instead, also made sure the VVVVVV still compiled on linux/windows/mac.

**Comment by andrewrk** [comment]
Nice! You have commit privileges right?

**Comment by marler8997** [comment]
Yup

**Pull request merged**